File: scala/examples/src/main/scala/stripe/Processor.scala
package stripe


class Processor {

  def process(acceptedLanguages: String, supportedLanguages: List[String]): List[String] = {
    acceptedLanguages.replace(" ", "").split(",").toStream
      .filter(supportedLanguages.contains(_))
      .distinct
      .toList
  }

}

File: prep/interview-strip/src/main/java/Processor.java
import org.apache.commons.lang3.StringUtils;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Processor {

    public List<String> process(String acceptedLanguages, List<String> supportedLanguages) {

        return Arrays.asList(StringUtils.deleteWhitespace(acceptedLanguages).split(","))
            .stream().distinct().filter(supportedLanguages::contains).collect(Collectors.toList());

    }

}
